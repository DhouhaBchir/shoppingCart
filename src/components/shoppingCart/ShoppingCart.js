import React, { Component } from 'react';
import '../App.css';
import ProductsCartList from './ProductsCartList';
import {Link} from 'react-router-dom';
class ShoppingCart extends Component {
// <Link to={`/cart_details`} className="link">My Cart</Link>
    render() {
        let total = this.props.productsList.reduce(function (a,b) {
            if(b.in_cart) return (a + b.price);
            else return a;
        }, 0);
        return (
            <div className="shopping_cart">
                <ProductsCartList productsList={this.props.productsList} removeFromCart={this.props.removeFromCart}/>
                <p>Total: {total}</p>
            </div>
        );
    }
}

export default ShoppingCart;
