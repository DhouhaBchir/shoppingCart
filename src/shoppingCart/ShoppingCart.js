import React, { Component } from 'react';
import ProductsCartList from './ProductsCartList';

class ShoppingCart extends Component {
    render() {
        let total = this.props.productsList.reduce(function (a,b) {
            if(b.in_cart) return (a + b.price);
            else return a;
        }, 0);
        return (
            <div className="shopping_cart">
                <ProductsCartList productsList={this.props.productsList} removeFromCart={this.props.removeFromCart}/>
                <p>Total: {total}</p>
            </div>
        );
    }
}

export  default  ShoppingCart;
