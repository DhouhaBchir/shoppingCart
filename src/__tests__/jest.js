import React from 'react';
import renderer from 'react-test-renderer';
import ShoppingCart from '../shoppingCart/ShoppingCart';

describe('ShoppingCart (Snapshot)', () => {
    let _products =
        [{
            id: 1,
            name: "Product 1",
            price: 5,
            logo: "product.png",
            in_cart: true
        },
        {
            id: 2,
            name: "Product 2",
            price: 5,
            logo: "product.png",
            in_cart: true
        }];
    let removeFromCart = function (id) {
        console.log("id, ", id);
    };
    it('Login Page renders Login', () => {
        const component = renderer.create(<ShoppingCart productsList={_products} removeFromCart={removeFromCart}/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});

