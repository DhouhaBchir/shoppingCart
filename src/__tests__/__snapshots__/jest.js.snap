// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart (Snapshot) Login Page renders Login 1`] = `
<div
  className="shopping_cart"
>
  <div>
    <div
      className="product_cart_item"
    >
      <img
        className="product_cart_logo"
        src="/src/product.png"
      />
      <p
        className="product_cart_name"
      >
        Product 1
      </p>
      <p>
        5
        $
      </p>
      <button
        className="product_cart_remove"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="product_cart_item"
    >
      <img
        className="product_cart_logo"
        src="/src/product.png"
      />
      <p
        className="product_cart_name"
      >
        Product 2
      </p>
      <p>
        5
        $
      </p>
      <button
        className="product_cart_remove"
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>
  <p>
    Total: 
    10
  </p>
</div>
`;
